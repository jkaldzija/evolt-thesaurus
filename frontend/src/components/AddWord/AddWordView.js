import React from 'react';
import {Button, Card} from "reactstrap";
import ThemeButton from "../Form/ThemeButton/ThemeButton";
import AddWordDialog from "./AddWordDialog";
import style from "./style.module.css";

class AddWordView extends React.Component {

    state = {
        addSynonymDialogOpen: false
    };

    toggleDialog = (key) => () => {
        const fieldKey = `${key}DialogOpen`;
        this.setState({ [fieldKey]: !this.state[fieldKey] });
    };


    componentDidMount() {
    }

    componentDidUpdate() {
    }

    render() {

        return <Card>
            <h5 className="mb-4">Know a new word?</h5>
            <ThemeButton onClick={this.toggleDialog("addSynonym")} className={style['add-word']}>Let us know</ThemeButton>
            <AddWordDialog
                toggle={this.toggleDialog("addSynonym")}
                isOpen={this.state.addSynonymDialogOpen}
            />
        </Card>
    };

}

export default AddWordView;
