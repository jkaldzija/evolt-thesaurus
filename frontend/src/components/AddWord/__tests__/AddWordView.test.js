import React from 'react';
import {shallow} from "enzyme";
import AddWordView from "../AddWordView";

describe('Add Word Dialog', function () {
    it('should render correctly', function () {
        const component = shallow(
            <AddWordView />
        );
        expect(component).toMatchSnapshot();
    });

    it('should toggle dialog properly', function (){
        const word = {
            value: "word",
            synonyms: null
        };
        const component = shallow(
            <AddWordView />
        );

        component.instance().toggleDialog("addSynonym")();
        expect(component.state("addSynonymDialogOpen")).toEqual(true);
        component.instance().toggleDialog("addSynonym")();
        expect(component.state("addSynonymDialogOpen")).toBe(false);
    });

});
