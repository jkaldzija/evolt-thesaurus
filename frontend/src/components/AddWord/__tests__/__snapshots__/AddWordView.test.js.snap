// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Word Dialog should render correctly 1`] = `
<Card
  tag="div"
>
  <h5
    className="mb-4"
  >
    Know a new word?
  </h5>
  <ThemeButton
    className="add-word"
    onClick={[Function]}
  >
    Let us know
  </ThemeButton>
  <AddWordDialog
    isOpen={false}
    open={false}
    reloadData={[Function]}
    title="Add word"
    toggle={[Function]}
    word={
      Object {
        "value": "",
      }
    }
  />
</Card>
`;
